docs/source/raft_configuration.md

```diff
@@ -330,5 +330,34 @@ channels (this does not mean they share data --- they do not!), if a Raft node i
 part of the consenter set in many channels, you might want to lengthen the amount
 of time it takes to trigger an election to avoid inadvertent leader elections.
 
+2. As aforementioned, write-ahead logs (wal) and snapshots should be persisted in
+order to attain safety property of Raft protocol. However, it may happen if a node
+loses its data by accident, and the node would panic with following message:
+```
+tocommit(X) is out of range [lastIndex(Y)]. Was the raft log corrupted, truncated, or lost?
+```
+Manual operation would be required to resurrect problematic node in this case.
+* Safe recovery:  
+The only way to handle the corruption or loss of a WAL is to remove the corrupted
+consenter from the quorum, and add a new one in its place, which can be done following
+instructions in reconfiguration section in this doc. This is the most appropriate
+option, as it does not violate safety property of Raft protocol. However, this is heavy
+operation and affects all channels this node participates in.
+* Unsafe recovery:  
+It is possible to recover from a WAL corruption without removing the consenter from the
+consenter set and re-adding it. However, these procedures violate the Raft protocol safety
+guarantees and, in certain unlikely cases, could result in a blockchain fork. To minimize
+these possibilities, ensure that your cluster has quorum on all channels for a few minutes
+before proceeding.
+  * Option 1: wipe out wal and snapshot directories of problematic channel from the corrupted
+node, restart current leader node, then start corrupted orderer. This essentially treats
+corrupted orderer as a fresh node without any data. This is the simplest option that requires
+restarting leader node, which often leads to system downtime.
+  * Option 2: to avoid downtime caused by restarting leader node, you could also wipe out
+wal and snapshot directories of problematic channel from the corrupted node, and copy
+a snapshot from an alive node with greater index than `X` in above panic message. The index
```

Comment:
Should the snapshot be copied only from the leader node here?
Besides, out of this topic, it might be a good idea to have orderer provided some command to rollback its local ledger to some height and then re-track, like the peer does today.

Reply:
it can be copied from either leader or other followers. as for orderer ledger rollback, i believe it's addressing a different problem other than corrupted WAL. But since we are here, what use cases do you have in mind?

Reply:
> it might be a good idea to have orderer provided some command to rollback its local ledger to some height and then re-track, like the peer does today.


Fabric guarantees Block finality. We can't revert blocks, otherwise peers would get forked.